solarforecastarbiter/validation/tasks.py

```diff
@@ -533,3 +613,43 @@ def daily_observation_validation(access_token, start, end, base_url=None):
             logger.warning(('Skipping daily validation of %s '
                             'not enough values'), observation.name)
             continue
+
+
+def apply_validation(observation, observation_values):
+    """
+    Applies the appropriate daily or immediate validation functions to the
+    observation_values depending on the length of the data.
+
+    Parameters
+    ----------
+    observation : solarforecastarbiter.datamodel.Observation
+    observation_values : pandas.DataFrame
+        Must have 'value' and 'quality_flag' columns
+
+    Returns
+    -------
+    pandas.DataFrame
+        With the same index as the input and 'quality_flag' updated
+        appropriately
+
+    Raises
+    ------
+    TypeError
+        If the supplied observations_values is not a DataFrame with a
+        DatetimeIndex
+    """
+    data = observation_values.sort_index()
+    if (
+            not isinstance(data, pd.DataFrame) or
+            not isinstance(data.index, pd.DatetimeIndex)
+    ):
+        raise TypeError('Expect observation_values to have a DatetimeIndex')
+    if data.empty:
+        return data
+    if (
+            (data.index[-1] - data.index[0]) > pd.Timedelta('1d') and
+            (len(data['value'].dropna()) > 10)
```

Comment:
Could you get away with only the first test if you used `data.first_valid_index()` and `data.last_valid_index()`?

Reply:
No, you could have only two valid indices a day apart and still not have 10 valid points which will raise an error for daily validation